sctk/src/application.rs

```diff
@@ -628,17 +637,114 @@ where
                             state.set_scale_factor(factor);
                         }
                     }
+                    SctkEvent::DataSource(DataSourceEvent::DndFinished) | SctkEvent::DataSource(DataSourceEvent::DndCancelled)=> {
+                        surface_ids.retain(|id, surface_id| {
+                            match surface_id {
+                                SurfaceIdWrapper::Dnd(inner) => {
+                                    drop(compositor_surfaces.remove(&inner));
+                                    interfaces.remove(inner);
+                                    states.remove(inner);
+                                    destroyed_surface_ids.insert(id.clone(), *surface_id);
+                                    false
+                                },
+                                _ => true,
+                            }
+                        })
+                    }
+                    _ => {}
                 }
             }
+            IcedSctkEvent::DndSurfaceCreated(wl_surface, dnd_icon) => {
+                // if the surface is meant to be drawn as a custom widget by the
+                // application, we should treat it like any other surfaces
+                //
+                // if the surface is meant to be drawn by a widget that implements
+                // draw_dnd_icon, we should mark it and not pass it to the view method
+                // of the Application
```

Comment:
Is this accurate for the current implementation? It seems to be calling `application.view(id)` the same way for either `DndIcon::Custom` or `DndIcon::Widget`.

Reply:
Ahh ya that comment is out of date. I wanted to not pass it to the view method at all, but it turned out that I had to do it the way it's currently done

Reply:
Okay. If it's something we'd like to improve in the future, it could be changed to a `TODO` comment.

If we find a good solution here, that should also make popups more ergonomic to handle. Perhaps something we could create an Iced RFC for.